asv_bench/benchmarks/benchmarks.py

```diff
@@ -148,6 +193,38 @@ def time_merge(self, shapes, how, sort):
         )
 
 
+class TimeMerge2:
+    param_names = ["shapes", "sort"]
+    params = [
+        get_benchmark_shapes("TimeMerge2"),
+        [True, False],
+    ]
+
+    def setup(self, shapes, sort):
+        self.df1 = generate_dataframe("int", *shapes[0], RAND_LOW, RAND_HIGH)
+        self.df2 = generate_dataframe("int", *shapes[1], RAND_LOW, RAND_HIGH)
+        self.rows_count = int(0.25 * shapes[1][0])
+
+    def time_merge_2intkey(self, shapes, sort):
+        execute(IMPL.merge(self.df1, self.df2, sort=sort))
+
+    def time_merge_dataframe_empty_right(self, shapes, sort):
+        execute(IMPL.merge(self.df1, self.df2.iloc[:0], sort=sort))
```

Comment:
`iloc` is another Modin operation. Not sure we want to measure `merge` along with `iloc`.

Reply:
This function for an empty `:0` should work very fast, we can leave it like that.

Reply:
where needed